### PR TITLE
Deprecate `conda.plan`

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Moves functions still in use (to where they are used):
- `conda.plan._get_best_prec_match` → `conda.misc._get_best_prec_match`
- `conda.plan.revert_actions` → `conda.cli.install.revert_actions`

Deprecates unused functions:
- `conda.plan.print_dists`
- `conda.plan.display_actions`
- `conda.plan.add_unlink`
- `conda.plan.add_defaults_to_specs`
- `conda.plan.execute_actions`
- `conda.plan._plan_from_actions`
- `conda.plan._inject_UNLINKLINKTRANSACTION`
- `conda.plan._handle_menuinst`
- `conda.plan.install_actions`
- `conda.plan.get_blank_actions`
- `conda.plan` as an entrypoint

Extends #13869
Resolves https://github.com/conda/conda/issues/12421

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: conda/misc.py [hunk 1]
```diff
@@ -341,3 +344,18 @@ def clone_env(prefix1, prefix2, verbose=True, quiet=False, index_args=None):
...
+    for chn in context.channels:
+        channel_matcher = ChannelMatch(chn)
+        prec_matches = tuple(
+            prec for prec in precs if channel_matcher.match(prec.channel.name)
+        )
+        if prec_matches:
+            break
+    else:
+        prec_matches = precs
+    log.warn("Multiple packages found:%s", dashlist(prec_matches))
```
Comment (jezdez) [inline conda/misc.py:360]:
```suggestion
    log.warn("Multiple packages found: %s", dashlist(prec_matches))
```

File: conda/misc.py [hunk 2]
```diff
@@ -341,3 +344,18 @@ def clone_env(prefix1, prefix2, verbose=True, quiet=False, index_args=None):
         index_args=index_args,
     )
     return actions, untracked_files
+
+
+def _get_best_prec_match(precs):
+    assert precs
+    for chn in context.channels:
+        channel_matcher = ChannelMatch(chn)
```
Comment (jezdez) [inline conda/misc.py:352]:
```suggestion
    for channel in context.channels:
        channel_matcher = ChannelMatch(channel)
```

File: conda/cli/install.py [hunk 3]
```diff
@@ -444,6 +454,45 @@ def install(args, parser, command="install"):
...
+            link_precs.add(precs[0])
+
+    if not_found_in_index_specs:
+        raise PackagesNotFoundError(not_found_in_index_specs)
+
+    final_precs = IndexedSet(PrefixGraph(link_precs).graph)  # toposort
+    unlink_precs, link_precs = diff_for_unlink_link_precs(prefix, final_precs)
+    stp = PrefixSetup(prefix, unlink_precs, link_precs, (), user_requested_specs, ())
+    txn = UnlinkLinkTransaction(stp)
+    return txn
```
Comment (jezdez) [inline conda/cli/install.py:493]:
```suggestion
    setup = PrefixSetup(prefix, unlink_precs, link_precs, (), user_requested_specs, ())
    return UnlinkLinkTransaction(setup)
```

**Review by jezdez — APPROVED** [review]
Approved with comments

File: conda/misc.py (hunk 1 above)
Comment (kenodegard) [reply conda/misc.py:360]: FYI the space is usually skipped since `dashlist` starts with a newline character

**Review by beeankha — APPROVED** [review]

**Pull request merged**